src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -933,12 +939,12 @@ public bool TryGetSingle(out float value)
         /// </exception>
         public float GetSingle()
         {
-            if (TryGetSingle(out float value))
```

Comment:
Does the new pattern have better IL? It seems like leaving the actual `throw` at the end of the method as it was previously should be fine.

Reply:
It's shaving off one `throw` instruction, although interestingly the assembly is identical in this case:

https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgYGYAE9MGFMDeqmJmxpGmARgPY0SYAqATgJ4DiApsAMpgB2AcwicAFDQCuwTADMINAIbSAbgogTOASkwBeAHyZgAC2Y0A7pn6cLAORrAAkgFsADiKed+wTgBMAogAeAMacLsBgNPyimgDc5CTx2FgwCABs2AAsTCbmgVHa+oY5FlzAedFxKKRJ2GmYgSFhEfyYpeUFBsamFlYWDvyqEGA+APIunMxKzQ2h4ZEViYmUcorSpXxCIozFABKcEOPM0YlEVdWkYDKYogCELBzcG8JiktIrSpiDGpqaidWn50B2W67UqQMwAF8/qRoSQYAB2T5qDRg0hQs4JDE1d5rR4CZ79IZWbbdY5YgFAy7Xe7rfEicRSWTyD5fLS/LH/WHnBFI9ScVGA9HgrldcytbigxLoiFAA===

Reply:
OK; I suppose it makes sense to just keep the changes in this PR just for consistency and thus having to keep in mind only 1 pattern instead of 2.

Reply:
On second thought, the happy path does do an extra short jump in the IL, so in theory it could be slower. However, since the assembly is the same I think the changes are fine.

Reply:
> It's shaving off one throw instruction, although interestingly the assembly is identical in this case:

The assembly is identical since you have a throwing placeholder for the `TryGetSingle` method in your test. If you change the `TryGetSingle` method to do something useful, the assembly will be different for the two cases.

> the happy path does do an extra short jump in the IL,

The JIT is smart enough to recognize always throwing paths and reorder the method blocks as necessary.